core/src/main/scala/org/apache/spark/deploy/Client.scala

```diff
@@ -176,6 +190,25 @@ private class ClientEndpoint(
       } else if (!Utils.responseFromBackup(message)) {
         System.exit(-1)
       }
+
+    case DriverStatusResponse(found, state, _, _, _) =>
+      if (found) {
+        state.get match {
+          case DriverState.FINISHED | DriverState.FAILED |
+               DriverState.ERROR | DriverState.KILLED =>
+            logInfo(s"State of $submittedDriverID is ${state.get}, " +
+              s"exiting spark-submit JVM.")
+            System.exit(0)
+          case _ =>
+            Thread.sleep(REPORT_DRIVER_STATUS_INTERVAL)
+            logInfo(s"State of $submittedDriverID is ${state.get}, " +
+              s"continue monitoring driver status.")
+            asyncSendToMasterAndForwardReply[DriverStatusResponse](
```

Comment:
When you submit too much applications at the same time, this could lead to heavy communication burden to the driver. I would suggest check less frequent (like increasing the interval to 10s)

Reply:
@jiangxb1987 Thanks for reviewing. I have changed it to 10 seconds and took care of your other comments. Kindly review the PR again.